source/extensions/injected_credentials/generic/generic_impl.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "source/common/config/datasource.h"
+#include "source/extensions/injected_credentials/common/credential.h"
+#include "source/extensions/injected_credentials/common/secret_reader.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace InjectedCredentials {
+namespace Generic {
+
+/**
+ * Implementation of credential injector's interface.
+ */
+class GenericCredentialInjector : public Common::CredentialInjector {
+public:
+  GenericCredentialInjector(const std::string& header, Common::SecretReaderSharedPtr secret_reader)
```

Comment:
I think SecretReadSharedPtr could be SecretReaderConstSharedPtr, or `std::shared_ptr<const SecretReader>`.